src/rpc.rs

```diff
@@ -367,3 +371,54 @@ pub fn get_id_specific_size(num: Block) -> HashMap<u32, u32> {
 	}
 	index
 }
+
+pub fn parse_urls(urls: Vec<String>) -> Result<Vec<url::Url>> {
+	urls.iter()
+		.map(|url| url::Url::parse(url))
+		.map(|r| r.map_err(|parse_error| anyhow!("Cannot parse URL: {}", parse_error)))
+		.collect::<Result<Vec<_>>>()
+}
+
+pub async fn check_connection(
+	full_node_ws: &[url::Url],
+) -> Option<WebSocketStream<MaybeTlsStream<TcpStream>>> {
+	// TODO: We are ignoring errors here, we should probably return result instead of option
+	for url in full_node_ws.iter() {
+		if let Ok((ws, _)) = connect_async(url).await {
+			return Some(ws);
+		};
+	}
+	None
+}
+
+pub async fn check_http(full_node_rpc: Vec<String>) -> Result<String> {
```

Comment:
We should make an effort to either create more descriptive function/variable names or add explanatory comments (ideally, both).

Reply:
Added some comments for the functions